packages/mrml-python/src/lib.rs

```diff
@@ -275,5 +264,6 @@ fn register(_py: Python<'_>, m: &Bound<'_, PyModule>) -> PyResult<()> {
     m.add_function(wrap_pyfunction!(local_loader, m)?)?;
     m.add_function(wrap_pyfunction!(http_loader, m)?)?;
     m.add_function(wrap_pyfunction!(memory_loader, m)?)?;
+    m.gil_used(false)?;
```

Comment:
No GIL!

Reply:
Could you explain me why?

Reply:
I'm not totally sure where your Python background is, so I'll start high-level. Skip whatever you already know though. :)

The [GIL](https://wiki.python.org/moin/GlobalInterpreterLock) is being removed (currently behind a compile flag for now) from Python. This means Python is going to have real parallelism via threads (Yay!), in contrast to before where only one Python thread could be running at a time (loosely speaking).

Old Python code in general will continue to run without any changes needed since all Python data structures are getting thread-safe locks put on them. The main issue comes from native Python extensions (like this one), which didn't need to be thread safe before. Python itself is unable to verify if a native extension is ready for this new GIL-less world, so native extensions must declare they are not relying on the GIL for correctness. If not all native extensions declare they are not using the GIL in a Python program, the interpreter will fall back to using the GIL.

That is what `gil_used(false)` is doing here. It's signaling to Python that this extension has been verified to work without the GIL. This flag does not affect behavior in older Python versions though.

However, I did look at the pyo3 [docs again](https://pyo3.rs/v0.23.3/free-threading.html), and it looks like the preferred way to declare this when using the pymodule macro is to put `#[pymodule(gil_used = false)]` in the macro and not imperatively like I did here.